Gems/AWSGameLift/Code/AWSGameLiftClient/Source/AWSGameLiftClientLocalTicketTracker.cpp

```diff
@@ -0,0 +1,168 @@
+        }
+    }
+
+    void AWSGameLiftClientLocalTicketTracker::ProcessPolling(
+        const AZStd::string& ticketId, const AZStd::string& playerId)
+    {
+        while (m_status == TicketTrackerStatus::Running)
+        {
+            auto gameliftClient = AZ::Interface<IAWSGameLiftInternalRequests>::Get()->GetGameLiftClient();
+            if (gameliftClient)
+            {
+                Aws::GameLift::Model::DescribeMatchmakingRequest request;
+                request.AddTicketIds(ticketId.c_str());
+
+                auto describeMatchmakingOutcome = gameliftClient->DescribeMatchmaking(request);
+                if (describeMatchmakingOutcome.IsSuccess())
+                {
+                    if (describeMatchmakingOutcome.GetResult().GetTicketList().size() == 1)
+                    {
+                        auto ticket = describeMatchmakingOutcome.GetResult().GetTicketList().front();
+                        if (ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::COMPLETED)
+                        {
+                            m_status = TicketTrackerStatus::Idle;
+                            AZ_TracePrintf(AWSGameLiftClientLocalTicketTrackerName,
+                                "Matchmaking ticket %s is complete.", ticket.GetTicketId().c_str());
+                            RequestPlayerJoinMatch(ticket, playerId);
+                            return;
+                        }
+                        else if (ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::TIMED_OUT ||
+                            ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::FAILED ||
+                            ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::CANCELLED)
+                        {
+                            m_status = TicketTrackerStatus::Idle;
+                            AZ_Error(AWSGameLiftClientLocalTicketTrackerName, false, "Matchmaking ticket %s is not complete, %s",
+                                ticket.GetTicketId().c_str(), ticket.GetStatusReason().c_str());
+                            return;
+                        }
+                        else if (ticket.GetStatus() == Aws::GameLift::Model::MatchmakingConfigurationStatus::REQUIRES_ACCEPTANCE)
+                        {
+                            // broadcast acceptance requires to player
+                        }
+                        else
+                        {
+                            AZ_TracePrintf(AWSGameLiftClientLocalTicketTrackerName, "Matchmaking ticket %s is processing, %s.",
+                                ticket.GetTicketId().c_str(), ticket.GetStatusReason().c_str());
+                        }
+                    }
+                    else
+                    {
+                        AZ_Error(AWSGameLiftClientLocalTicketTrackerName, false, "Unable to find expected ticket with id %s", ticketId.c_str());
```

Comment:
Should this error state end the poll?

Reply:
nope, as this error maybe transient

Reply:
IMHO suggest adding comment to code as to why this is not terminal for future maintainers